### PR TITLE
Added the option to set the content-type metadata when uploading a file

### PR DESCRIPTION
**Pull request by pdelagrave**
Signed-off-by: Ahmed Mohemed <amohemed@pivotal.io>

### CODE REVIEW

File: s3client.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ type S3Client interface {
 	BucketFiles(bucketName string, prefixHint string) ([]string, error)
 	BucketFileVersions(bucketName string, remotePath string) ([]string, error)
 
-	UploadFile(bucketName string, remotePath string, localPath string, acl string, serverSideEncryption string, kmsKeyId string) (string, error)
+	UploadFile(bucketName string, remotePath string, localPath string, acl string, contentType string, serverSideEncryption string, kmsKeyId string) (string, error)
```
Comment (vito) [inline s3client.go]: I think it's time to refactor this; the argument list keeps growing and growing. Maybe we could pass in an `UploadInput` with fields set and just have `UploadFile` fill in the `Body` attribute?

**Comment by vito** [comment]
Thanks! Building on concerns that I expressed in a previous (unrelated) PR though, the `UploadFile` function is getting a little hairy. Seven `string` arguments is just too sketchy and more keep getting added. Would you mind taking a swing at refactoring this?

**Comment by pdelagrave** [comment]
Hi!

Sure, that's a legit concern.

We need this feature so we can consider working on that refactor at some point, but it's not a priority and we have a satisfying resource docker image to work with at the moment.

**Comment by vito** [comment]
Sorry, I just can't continue to justify the growing tech debt in that area. Thanks though!

**Pull request closed without merging**